src/main/java/org/jitsi/utils/dsi/DominantSpeakerIdentification.java

```diff
@@ -431,6 +482,96 @@ private synchronized Speaker getOrCreateSpeaker(T id)
         return speaker;
     }
 
+    private synchronized void updateLoudestList(Speaker speaker, int level, long now)
+    {
+        long oldestValid = now - energyExpireTimeMs;
+
+        /* exponential smoothing */
+        speaker.energyScore = (energyAlphaPct * level + (100 - energyAlphaPct) * speaker.energyScore) / 100;
+
+        logger.debug("Want to add " + speaker.id.toString() + " with score " + speaker.energyScore + ". Last level = " + level + ".");
+
+        int i = 0;
+        while (i < loudest.size())
+        {
+            Speaker cur = loudest.get(i);
+            if (cur.getLastLevelChangedTime() < oldestValid)
+            {
+                logger.debug("Removing " + cur.id.toString() + ". old.");
```

Comment:
I feel like these should be `trace` rather than `debug` level logs - I generally prefer trace for logs that run on every operation.

String concatenation isn't *that* expensive, but I also wonder if it's worthwhile to switch to using `Logger2` and use closures to create the log messages, so we don't do the concatenation when the logs aren't enabled.

Reply:
For messages as simple as these it would be really better to use a logging framework that understands template parameters, e.g. Logback or log4j2 (both via slf4j) or [Flogger](https://github.com/google/flogger).
The statement would then be `logger.debug("Removing {} old", cur.id);`, which is essentially free if debug logging is disabled (and cheaper than lambdas). And the readability gain is also massive.

Reply:
I wanted to log a lot for testing. I can remove them before submitting.

Reply:
moved to logger2, used closures.

Reply:
move to trace level as well.